src/main/java/com/puppycrawl/tools/checkstyle/checks/annotation/MissingDeprecatedCheck.java

```diff
@@ -255,7 +255,7 @@ private boolean containsDeprecatedTag(DetailNode javadoc) {
     private static DetailAST getParent(DetailAST commentBlock) {
         DetailAST result = commentBlock.getParent();
 
-        if (result == null) {
+        if (result.getType() == TokenTypes.COMPILATION_UNIT) {
```

Comment:
`result.getType() == TokenTypes.COMPILATION_UNIT`
may be instead of repeating this comparison add new method to DetailAST (or TokenUtil) like `boolean isRootNode()` that will hide this and make it easier to read.
Method content will be as simple as `return this.type == TokenTypes.COMPILATION_UNIT`

what do you think?

Reply:
Done.